vllm/model_executor/models/utils.py

```diff
@@ -238,6 +238,90 @@ def load_weights(
         return autoloaded_weights
 
 
+class OffloadedTensorMode(TorchDispatchMode):
+
+    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
+        kwargs = {} if kwargs is None else kwargs
+        tensor = func(*args, **kwargs)
+
+        if (func is torch.ops.aten.empty.memory_format
+                and tensor.device != "cpu"):
+            global _CPU_OFFLOAD_BYTES, _CPU_OFFLOAD_MAX_BYTES
+            if _CPU_OFFLOAD_BYTES < _CPU_OFFLOAD_MAX_BYTES:
+                _CPU_OFFLOAD_BYTES += tensor.numel() * tensor.element_size()
+                return OffloadedTensor(tensor)
+        return tensor
+
+
+class OffloadedTensor(torch.Tensor):
+
```

Comment:
need to have `__new__` , similar to https://github.com/albanD/subclass_zoo/blob/ec47458346c2a1cfcd5e676926a4bbc6709ff62e/base_tensor.py#L11 

we might also need https://github.com/albanD/subclass_zoo/blob/ec47458346c2a1cfcd5e676926a4bbc6709ff62e/base_tensor.py#L25

Reply:
We do generally have support for subclasses that implement both torch_function and torch_dispatch, although if you only need torch_dispatch then I agree that you probably want to disable torch_function as linked above.

Let me know if you have any other questions / would like to chat more about the subclass work you're doing!